interface/patient_file/front_payment.php

```diff
@@ -401,6 +401,11 @@ function calcTaxes($row, $amount)
     WHERE encounter = ?", array($payrow['encounter']));
     $frow = $facilityService->getById($tmprow['facility_id']);
 
+ if($_REQUEST['radio_type_of_payment']=='pre_payment'){
+  // if pre_payment is being made then use the main office address added by Sherwin 7/21/2017
+   $frow = sqlQuery("SELECT * FROM `facility` WHERE name != '' LIMIT 1");
+   }
```

Comment:
actually there are nice service functions in services/FacilityService.php. check it out and see if any will work for you. For example, this may work for you:
``````
    if ($_REQUEST['radio_type_of_payment'] != 'pre_payment') {
        $tmprow = sqlQuery("
        SELECT facility_id
        FROM form_encounter
        WHERE encounter = ?", array($payrow['encounter']));
        $frow = $facilityService->getById($tmprow['facility_id']);
    } else {
        // if pre_payment is being made then use the main office address - added by Sherwin 7/21/2017
        $frow = $facilityService->getPrimaryBusinessEntity;
    }